File: apps/macos/Sources/MarkLeaf/Services/NewDocumentKind.swift
import Foundation

/// 新建文档的类型及其默认保存信息。
enum NewDocumentKind: Equatable {
    case markdown
    case plainText

    var fileExtension: String {
        switch self {
        case .markdown: return "md"
        case .plainText: return "txt"
        }
    }

    var editorDocumentType: String {
        switch self {
        case .markdown: return "markdown"
        case .plainText: return "plainText"
        }
    }

    var defaultFileName: String {
        "未命名.\(fileExtension)"
    }

    static func from(fileExtension: String?) -> NewDocumentKind {
        fileExtension?.lowercased() == "txt" ? .plainText : .markdown
    }
}

File: apps/windows/MarkLeaf/Documents/NewDocumentKind.cs
namespace MarkLeaf.Documents;

public enum NewDocumentKind
{
    Markdown,
    PlainText,
}

public static class NewDocumentKindExtensions
{
    public static string FileExtension(this NewDocumentKind kind)
        => kind == NewDocumentKind.PlainText ? "txt" : "md";

    public static string EditorDocumentType(this NewDocumentKind kind)
        => kind == NewDocumentKind.PlainText ? "plainText" : "markdown";

    public static string DefaultFileName(this NewDocumentKind kind)
        => kind == NewDocumentKind.PlainText ? "未命名.txt" : "未命名.md";

    public static NewDocumentKind FromExtension(string? extension)
        => string.Equals(extension?.TrimStart('.'), "txt", StringComparison.OrdinalIgnoreCase)
            ? NewDocumentKind.PlainText
            : NewDocumentKind.Markdown;
}
